rule/deep_exit.go

```diff
@@ -76,5 +79,39 @@ func (w *lintDeepExit) Visit(node ast.Node) ast.Visitor {
 func (w *lintDeepExit) mustIgnore(fd *ast.FuncDecl) bool {
 	fn := fd.Name.Name
 
-	return fn == "init" || fn == "main" || (w.isTestFile && fn == "TestMain")
+	return fn == "init" || fn == "main" || w.isTestMain(fd) || w.isTestExample(fd)
+}
+
+func (w *lintDeepExit) isTestMain(fd *ast.FuncDecl) bool {
+	return w.isTestFile && fd.Name.Name == "TestMain"
+}
+
+// isTestExample returns true if the function is a testable example function.
+// See https://go.dev/blog/examples#examples-are-tests for more information.
+//
+// Inspired by https://github.com/golang/go/blob/go1.23.0/src/go/doc/example.go#L72-L77
+func (w *lintDeepExit) isTestExample(fd *ast.FuncDecl) bool {
+	if !w.isTestFile {
+		return false
+	}
+	if !isTest(fd.Name.Name, "Example") {
+		return false
+	}
+	return len(fd.Type.Params.List) == 0
```

Comment:
What about a function with a return value? like
```go
func ExampleFoo() int {
	return 0
}
```
Is it a test example?

Reply:
Nope

Reply:
> What about a function with a return value?

This case is rare and not covered even with [the Go's code](https://github.com/golang/go/blob/go1.23.0/src/go/doc/example.go#L51).

For example, my VS Code detects `ExampleFoo` as an example:
<img width="303" alt="image" src="https://github.com/user-attachments/assets/fb8f3563-d14e-49ea-9a66-85446a522c23">

and show the warning:

<img width="397" alt="image" src="https://github.com/user-attachments/assets/795a17b6-02a5-4875-b15e-6dfe2024afcf">

Do I need to cover it here?

Anyway, this mistake is covered by the [`tests`](https://pkg.go.dev/golang.org/x/tools@v0.27.0/go/analysis/passes/tests) analyzer.

Reply:
>This case is rare and not covered even with the Go's code

I seems that the [`playExample`](https://github.com/golang/go/blob/6885bad7dd86880be6929c02085e5c7a67ff2887/src/go/doc/example.go#L153) function is OK with examples returning values therefore ...

>Do I need to cover it here?

no